tests/test_repository.py

```diff
@@ -21,7 +21,17 @@
 from twine import utils
 
 
+@pytest.fixture()
+def repo():
+    return repository.Repository(
+        repository_url=utils.DEFAULT_REPOSITORY,
+        username="username",
+        password="password",
+    )
+
+
 def test_gpg_signature_structure_is_preserved():
+    """Test that gpg signature structure doesn't change"""
```

Comment:
Personally, I find the docstrings in this file to be redundant with the test function name, and would prefer to leave them out. A quick scan suggests that's the de-facto pattern in the rest of the tests.

@sigmavirus24 What do think?

Reply:
@bhrutledge I find that docstrings are better to have in general than not. I do find that most people struggle with writing them for tests though, and so if someone isn't going to expand upon the reasoning for the test (what are we trying to prevent from breaking) then they can seem valueless. Regardless, I would love for us to get to a place where we can enforce everything (that's public) having a docstring with `flake8-docstrings`. This gets us there, even if they're temporarily redundant.

Reply:
In that case, I will keep the docstrings as is and won't remove them

Reply:
Downstream packagers insist on having tests included in the package thus everyone who downloads Twine is getting a copy of them (even if they don't want them and have internet that makes it painful to download them). So in a way, everyone is getting copies of our tests and that makes them more public-ish. Further, those downstream packagers also run our tests and if one of them fails and the test name makes no sense to them because they have no context on the project, a docstring that explains the purpose of the test and the intent behind it can help them take action and debug more before they have to involve you or me. Less work for us is generally a good thing.

Additoinally, as people write new tests, they tend to look for old ones for inspiration and patterns. Expecting everyone to understand immediately (or sit there and think about it long enough to understand) _what_ the purpose of the test is, is unreasonable and disrespectful to contributors time. Finally, people evaluating whether to use Twine or not might skim our tests. Understanding what exactly we're testing will give them greater confidence in the quality of the tool.

Reply:
Good points, and no argument from me on the value of explanatory docstrings. I was more concerned about new tests adopting the pattern of redundant docstrings, but that's something we can address when we get there.

So, no change required for now, and maybe we can come up with a holistic docstring guide in the future.

Reply:
I will probably come up with PR(s) in the future to add docstrings/comments to tests/functions which don't have that so that we can have more focussed discussion on the content of those and whether they actually explain the function and code. 

Also apologies for being the source of disagreement which my docstring changes. I will keep the future changes more focussed. 😞

Reply:
No worries, @deveshks. As a relatively new maintainer, I find these sorts of conversations valuable.